test/CMakeLists.txt

```diff
@@ -306,9 +307,11 @@ if(MIOPEN_TEST_MLIR)
     set(IMPLICITGEMM_MLIR_ARGS_W ${IMPLICITGEMM_ARGS} --verbose --disable-forward --disable-backward-data)
     
     add_custom_test(test_conv_igemm_mlir ALLOW_NONXDLOPS
-        COMMAND ${IMPLICITGEMM_MLIR_ENV_F} $<TARGET_FILE:test_conv2d> ${IMPLICITGEMM_MLIR_ARGS_F} --input 64 1024 14 14 --weights 1024 1024 1 1 --pads_strides_dilations 0 0 1 1 1 1
-        COMMAND ${IMPLICITGEMM_MLIR_ENV_F} $<TARGET_FILE:test_conv2d> ${IMPLICITGEMM_MLIR_ARGS_F} --input 64 256 56 56  --weights  256  256 1 1 --pads_strides_dilations 0 0 1 1 1 1
-        COMMAND ${IMPLICITGEMM_MLIR_ENV_F} $<TARGET_FILE:test_conv2d> ${IMPLICITGEMM_MLIR_ARGS_F} --input 64 128 58 58  --weights  128  128 3 3 --pads_strides_dilations 0 0 1 1 1 1
+        COMMAND ${IMPLICITGEMM_MLIR_ENV_F_CPP} $<TARGET_FILE:test_conv2d> ${IMPLICITGEMM_MLIR_ARGS_F} --input 64 1024 14 14 --weights 1024 1024 1 1 --pads_strides_dilations 0 0 1 1 1 1
+        COMMAND ${IMPLICITGEMM_MLIR_ENV_F_CPP} $<TARGET_FILE:test_conv2d> ${IMPLICITGEMM_MLIR_ARGS_F} --input 64 256 56 56  --weights  256  256 1 1 --pads_strides_dilations 0 0 1 1 1 1
+        COMMAND ${IMPLICITGEMM_MLIR_ENV_F_CPP} $<TARGET_FILE:test_conv2d> ${IMPLICITGEMM_MLIR_ARGS_F} --input 64 128 58 58  --weights  128  128 3 3 --pads_strides_dilations 0 0 1 1 1 1
+        COMMAND ${IMPLICITGEMM_MLIR_ENV_F_BIN} $<TARGET_FILE:test_conv2d> ${IMPLICITGEMM_MLIR_ARGS_F} --input 64 128 58 58  --weights  128  128 3 3 --pads_strides_dilations 0 0 1 1 1 1
+        COMMAND ${IMPLICITGEMM_MLIR_ENV_F_BIN} $<TARGET_FILE:test_conv2d> ${IMPLICITGEMM_MLIR_ARGS_F} --input 128 8 32 32  --weights  128  8 3 3 --pads_strides_dilations 0 0 1 1 1 1
```

Comment:
I didn't find any limitations for pads, so please add a test with non-zero padding.

Reply:
I'd prefer not add additional test for now because:
- MLIR is an experimental option now, it does not impact MIOpen general correctness.
- The work in MIOpen is barely pass-through to MLIR. This PR aims at just test that pass-through instead of correctness. 
-  Right now MLIR is working through a fair amount of kernel issues, it is most likely to expose issues if I'm to add more test at this specific commit
   - Even if we find a kernel crashing because of MLIR, there's no way from me to address it from MIOpen end. However, I don't think this should block the pass-through logic going into MIOpen
   - Once MLIR work through all its kernel generation issues, I can just bump the commit id for MIOpen to adopt the fix.

Reply:
But I agree that if we ever decide to drop `-MIOPEN_USE_MLIR` and make it the default compilation method, we should add more test coverage to make sure it can handle all situations.

Reply:
Oh, ok, I forget that this is experimental.

But the problem is that it still can block MIOpen CI.
Example: 
As i understand MLIR is working only for convolution, and for now it is applicable for a wide range of convolutions.

MIOpen CI tests MLIR in stage('Fp32 Hip MLIR') which includes several convolution tests without MIOPEN_DEBUG_FIND_ONLY_SOLVER, and this creates conditions for MLIR execution in one of these tests.

Reply:
So we can reduce number of test to test for stage('Fp32 Hip MLIR'), to test  only MLIR and only for required cases.
Or to change IsApplicable range.

Reply:
For me, the first is better.

Reply:
I can't use github suggestion for this, but:
At line:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/6128704ed8a325474868a42d8853139f0defa971/test/CMakeLists.txt#L91

write:
```
if(MIOPEN_TEST_MLIR)
    set(SKIP_ALL_EXCEPT_TESTS test_conv_igemm_mlir)
endif()
```
.
And line:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/6128704ed8a325474868a42d8853139f0defa971/test/CMakeLists.txt#L235
Change to 
```
        if( 
            ( NOT PARSE_SKIP_UNLESS_ALL OR MIOPEN_TEST_ALL)
            AND
            (NOT MIOPEN_TEST_MLIR OR (NAME IN_LIST SKIP_ALL_EXCEPT_TESTS))
        )
```

Reply:
@shurale-nkn I see what you mean. I think a way to address this is to update the Jenkins file such that instead of `make check` replace it with `make test_conv_igemm_mlir`?

From a different perspective though. I feel like knowing that MLIR build not only MLIR path work can be beneficial from the test coverage perspective? After all, once I shrink the test coverage to MLIR build only, MIOpen lose track of whether everything else work with MLIR build.

On other thought I have, regarding about handling the possibility of flakiness in the `Fp32 Hip MLIR` stage, is to handle it from jenkinsfile:

```jenkinsfile
stage('Fp32 Hip MLIR') {
   try {
     buildHipClangJob(...)
   } catch(error) {
     echo "MLIR build flakiness observed, retrying"
     retry(2) {
        buildHipClangJob(...)
     }
   } finally {
      reboot()
   }
}
```

Reply:
@jerryyin make test_conv_igemm_mlir is also posible way.

`"From a different perspective though. I feel like knowing that MLIR build not only MLIR path work can be beneficial from the test coverage perspective?"`

In that case we came to MIOpen CI blocker. The problem is, we don't know when it might show up. Better to use only tested solutions. Otherwise, someone will have to solve it later.
Convolution is pretty much isolated from another parts of MIOpen.

`"On other thought I have, regarding about handling the possibility of flakiness in the Fp32 Hip MLIR stage, is to handle it from jenkinsfile:"`
If retry (2) contains the same tests, we will have problems with correctness, which appear with some probability and we will have to restart the stage sometimes. In one form or another, we have already been through this, and it does not lead to anything good. Who know how many restarts it take.
If Test marked as failed and CI should ignore this, what's the point of this test?

Better to do full testing when MLIR will be ready, with true full applicable.
Because this is MIOpen CI :) to test something no need to merge it to develop :)

Reply:
@shurale-nkn Make sense, better safe than sorry is a better approach here. Also, agreed that jenkins retry is not a very elegant solution.

I appreciate your very detailed instructions in updating the test cmakefiles. I decided not to do that just so that I myself can do `make check` in my dev process to run through the whole set of tests and make sure everything is as expected.

Reply:
If MLIR needs some kind of periodic testing in the future, we can create a separate work on Jenkins.

Reply:
@shurale-nkn Thanks for reminding. MLIR already has a number of integration/nightly testing. We are in that "bug fixing" stage for outstanding issues. Hopefully once those issues are caught, I can add more tests in MIOpen end as well.

Reply:
@shurale-nkn @jerryyin 
> In that case we came to MIOpen CI blocker.

If this happens, we will disable MLIR testing stage in the Jenkinsfile until the problem is fixed. Let's KISS.